### ISSUE TITLE
USWDS Site - H4 headings appearing above H1

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds-site#2256]
### Describe the bug

On 139 USWDS pages, the H4 heading on the top right titled "on this page" appears above the H1. This violates [WCAG success criteria 2.4.6 headings and labels](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-descriptive.html).

### Steps to reproduce the bug

1. Go to any page with "on this page" in-page navigation. Example: https://designsystem.digital.gov/documentation/migration/
2. Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to inspect the page structure
3. See resulting page hierarchy:
![image](https://github.com/uswds/uswds-site/assets/117778073/a14d2c5e-e76b-4c7f-b322-abaa60a3be1c)

### Expected Behavior

The H1 should appear first in order on every page. In this case, the "Why migrate to USWDS 3.0?" should be first on the page. 

### System setup

This occurs in all browsers and operating systems.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.

**Comment by mejiaj** [comment]
**Todo**
- [x] Confirm if we're following implementation guidance. Content must precede.
- [x] Move this issue over to USWDS repo.
